crates/generated_parser/src/ast_builder.rs

```diff
@@ -4521,9 +4578,33 @@ impl<'alloc> AstBuilder<'alloc> {
         context.check_identifier_reference(token, &self.atoms.borrow())
     }
 
-    // Check Early Error for LabelIdentifier.
-    fn on_label_identifier(&self, token: &arena::Box<'alloc, Token>) -> Result<'alloc, ()> {
+    // Check Early Error for LabelIdentifier and note binding info to the
+    // stack
+    fn on_label_identifier(&mut self, token: &arena::Box<'alloc, Token>) -> Result<'alloc, ()> {
         let context = IdentifierEarlyErrorsContext::new();
+
+        let name = token.value.as_atom();
+        let offset = token.loc.start;
+
+        if let Some(info) = self.bindings.last() {
+            debug_assert!(info.offset < offset);
+        }
+
+        // Labels are not usually considered bindings, but we are using bindings
+        // in order to track duplicate label information. See `check_labelled_statement` for
+        // more information.
+        //
+        // If the label is associated with a continue or break statement, this information
+        // is popped from the stack. See `continue_statement` and `break_statement` for more
+        // information.
```

Comment:
what is "this information" ?
it sounds like `BindingInfo` is popped if continue/break exists, that doesn't sound right behavior.

Reply:
The short answer is -- yes, we pop the BindingInfo associated with a Break/Continue. We don't pop it if it is associated with a LabelledStatement. 

We can't tell at the call site if a labelIdentifier is part of a LabelStatement or if it is part of a Continue/BreakStatement. So, we push to binding with both. But, if we then find that the LabelIdentifier is part of a  continue/breakStatement, we remove it from bindings. It is the next step after pushing it. 

There are two other approaches -- change the grammar (which.. i don't know if it should be done frequently, I have been putting off changing the grammar until this is refactored as part of the context stack. All of this will be removed at that point (in #230 ), or mark a continue/break statement label as such. If we do the latter, that information will not be used again. 

Should the comment be updated or do you think the behavior needs to change? I would argue for the current solution.

Reply:
sorry, I misunderstood what "associated" means here.

(I was thinking that the comment is still talking about the case when this label is for labelled statement, and "when there's labelled break/continue refers this label")

so, the behavior is fine here.